### PR TITLE
issue/1331-shipping-provider-illegal-state

### LINKED ISSUE #1331: Woo: IllegalStateException Not a JSON Object: "failed"
User is experiencing a crash every time they switch to the orders tab, apparently due to invalid JSON in the response for the shipment provider list.

```
java.lang.IllegalStateException: Not a JSON Object: "failed"
    at com.google.gson.JsonElement.getAsJsonObject(JsonElement.java:91)
    at org.wordpress.android.fluxc.network.rest.wpcom.wc.order.OrderRestClient.jsonResponseToShipmentProviderList(OrderRestClient.kt:725)
    at org.wordpress.android.fluxc.network.rest.wpcom.wc.order.OrderRestClient.access$jsonResponseToShipmentProviderList(OrderRestClient.kt:49)
    at org.wordpress.android.fluxc.network.rest.wpcom.wc.order.OrderRestClient$fetchOrderShipmentProviders$request$1.invoke(OrderRestClient.kt:595)
    at org.wordpress.android.fluxc.network.rest.wpcom.wc.order.OrderRestClient$fetchOrderShipmentProviders$request$1.invoke(OrderRestClient.kt:49)
    at org.wordpress.android.fluxc.network.rest.wpcom.jetpacktunnel.JetpackTunnelGsonRequest$buildGetRequest$wrappedListener$1.onResponse(JetpackTunnelGsonRequest.kt:118)
    at org.wordpress.android.fluxc.network.rest.wpcom.jetpacktunnel.JetpackTunnelGsonRequest$buildGetRequest$wrappedListener$1.onResponse(JetpackTunnelGsonRequest.kt:89)
    at org.wordpress.android.fluxc.network.rest.GsonRequest.deliverResponse(GsonRequest.java:53)
    at com.android.volley.ExecutorDelivery$ResponseDeliveryRunnable.run(ExecutorDelivery.java:102)
    at android.os.Handler.handleCallback(Handler.java:789)
    at android.os.Handler.dispatchMessage(Handler.java:98)
    at android.os.Looper.loop(Looper.java:164)
    at android.app.ActivityThread.main(ActivityThread.java:6541)
    at java.lang.reflect.Method.invoke(Method.java)
    at com.android.internal.os.Zygote$MethodAndArgsCaller.run(Zygote.java:240)
    at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:767)
```

[Relevant Sentry link](https://sentry.io/organizations/a8c/issues/1140272336/?environment=release&project=1459556&query=is%3Aunresolved&statsPeriod=7d)

### PR DESCRIPTION
**Pull request by nbradbury**
Fixes #1331 - catches the exception that can occur when parsing the response to the shipment provider list. Note that I'll open a separate WCAndroid issue to show a toast when the list is empty.

### CODE REVIEW

File: .idea/codeStyles/Project.xml [hunk 1]
```diff
@@ -1,42 +1,10 @@
 <component name="ProjectCodeStyleConfiguration">
   <code_scheme name="Project" version="173">
-    <AndroidXmlCodeStyleSettings>
```
Comment (AmandaRiu) [inline .idea/codeStyles/Project.xml]: did you mean to check in these changes to the Project.xml file?

Comment (nbradbury) [reply .idea/codeStyles/Project.xml]: Nope. This seems to happen frequently with AS updates and I think it's fine to just accept the changes - they'll happen eventually anyway.

**Comment by nbradbury** [comment]
> This looks good to me! The only thing missing is a test in `MockedStack_WCOrdersTest` that tests this scenario

@AmandaRiu Added in f1b1a15.

**Comment by nbradbury** [comment]
@AmandaRiu I'm wondering if this should be changed so that the `OrderRestClient` catches the exception and responds with an error event, or simply responds with an error event when the provider list is empty. This really _is_ an error so that seems more appropriate.

**Comment by AmandaRiu** [comment]
> @AmandaRiu I'm wondering if this should be changed so that the OrderRestClient catches the exception and responds with an error event, or simply responds with an error event when the provider list is empty. This really is an error so that seems more appropriate.

@nbradbury I think that makes perfect sense. This way we can log it as an error event in the client and choose how to handle it in a more direct way.

**Comment by nbradbury** [comment]
@AmandaRiu I updated this to dispatch an error when parsing the providers fails. Ready for another look!

**Pull request merged**